common/attest_plugin.c

```diff
@@ -197,6 +198,86 @@ oe_result_t oe_unregister_verifier(oe_verifier_t* plugin)
     return _unregister_plugin(&verifiers, (oe_attestation_role_t*)plugin);
 }
 
+static oe_result_t _get_registered_plugin_format_ids(
+    struct plugin_list_node_t* list,
+    oe_uuid_t** format_ids,
+    size_t* format_ids_length)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    struct plugin_list_node_t* node = NULL;
+    size_t number_of_nodes = 0;
+    oe_uuid_t* ids = NULL;
+
+    if (!format_ids || !format_ids_length)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    *format_ids = NULL;
+    *format_ids_length = 0;
+
+    if (!list)
+        OE_RAISE(OE_NOT_FOUND);
+
+    // Calculate number of nodes
```

Comment:
Would it be possible to keep a local count? This would avoid having to compute the length every time.

Reply:
The linked list is not expected to go beyond a handful or so. Calculating the length is not expensive and avoid having to synchronize the count and the list.

---
In reply to: [365366521](https://github.com/openenclave/openenclave/pull/2402#discussion_r365366521) [](ancestors = 365366521)

Reply:
Maybe move this into a function?

Reply:
Discussed offline. Provide option to just get a count.